server/src/test/java/io/crate/analyze/CreateAlterPartitionedTableAnalyzerTest.java

```diff
@@ -204,231 +191,251 @@ public void testPartitionedByNestedColumns() {
-                "  index ft using fulltext(stuff, o['s']) with (analyzer='snowball')" +
-                ") partitioned by (name)");
-
+        assertThatThrownBy(() -> analyze(
+            "create table my_table(" +
+            "  name string index using fulltext," +
+            "  no_index string index off," +
+            "  stuff string," +
+            "  o object as (s string)," +
+            "  index ft using fulltext(stuff, o['s']) with (analyzer='snowball')" +
+            ") partitioned by (name)"
+        )).isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Cannot use column name with fulltext index in PARTITIONED BY clause");
     }
 
     @Test
     public void testPartitionedByCompoundIndex() {
-        expectedException.expect(IllegalArgumentException.class);
-        expectedException.expectMessage("Cannot use column ft with fulltext index in PARTITIONED BY clause");
-        analyze("create table my_table(" +
-                "  name string index using fulltext," +
-                "  no_index string index off," +
-                "  stuff string," +
-                "  o object as (s string)," +
-                "  index ft using fulltext(stuff, o['s']) with (analyzer='snowball')" +
-                ") partitioned by (ft)");
-
+        assertThatThrownBy(() -> analyze(
+            "create table my_table(" +
+            "  name string index using fulltext," +
+            "  no_index string index off," +
+            "  stuff string," +
+            "  o object as (s string)," +
+            "  index ft using fulltext(stuff, o['s']) with (analyzer='snowball')" +
+            ") partitioned by (ft)"
+        )).isExactlyInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Cannot use column ft with fulltext index in PARTITIONED BY clause");
     }
 
-    @Test(expected = IllegalArgumentException.class)
+    @Test
     public void testPartitionedByClusteredBy() {
-        analyze("create table my_table (" +
-                "  id integer," +
-                "  name string" +
-                ") partitioned by (id)" +
-                "  clustered by (id) into 5 shards");
+        assertThatThrownBy(() -> analyze(
+            "create table my_table (" +
+            "  id integer," +
+            "  name string" +
+            ") partitioned by (id)" +
+            "  clustered by (id) into 5 shards"
+        )).isExactlyInstanceOf(IllegalArgumentException.class);
     }
 
     @Test
     public void testAlterPartitionedTable() {
         BoundAlterTable analysis = analyze(
             "alter table parted set (number_of_replicas='0-all')");
-        assertThat(analysis.partitionName().isPresent(), is(false));
-        assertThat(analysis.isPartitioned(), is(true));
-        assertEquals("0-all", analysis.tableParameter().settings().get(AutoExpandReplicas.SETTING.getKey()));
+        assertThat(analysis.partitionName().isPresent()).isEqualTo(false);
```

Comment:
```suggestion
        `assertThat(analysis.partitionName()).isNotPresent();`
```

Reply:
minor: you could also check elsewhere and use `isPresent()` and `isNotPresent()`